weave/weave_types.py

```diff
@@ -784,7 +784,7 @@ def type_of_instance(cls, obj):
         if not obj:
             return cls(UnknownType())
         list_obj_type = TypeRegistry.type_of(obj[0])
-        for item in obj[1:]:
+        for item in obj[1:100]:
```

Comment:
WIP WIP WIP

Fix 4. Still need to properly sample (if this is even a good idea, given that this makes `List.type_of_instance` less reliable).